tmpl/context_funcs.go

```diff
@@ -206,3 +210,7 @@ func RequiredEnv(name string) (string, error) {
 
 	return "", fmt.Errorf("required env var `%s` is not set", name)
 }
+
+func SSM(key string) (val string, err error) {
+	return datasource.SSMGet(key)
```

Comment:
Not to say it is required in order to merge this, but just to note - I think we might want `datasource` to be in the `context` and separate per a context.

Alternatively it can be achieved by moving `datasource.PrepareAll(c)` to somewhere so that it runs once per each `helmfile.yaml` and its sub-helmfiles, assuming `helmfile` runs them one by one. `datasource` seems like a global state so moving it into the template context would be good to keep helmfile maintainable.

Reply:
@mumoshu this should probably be a blocker for the PR honestly. I only ever tested with 1 helmfile.

https://github.com/roboll/helmfile/pull/573/files#diff-94e9c44ce590540ca463d78859c012ebR56

The `ssmGetSpecs()` runs once to parse the helmfile to get the `ssm` definition from the root of the yaml

```yaml
ssm:
  - name: ssm1
    prefix: /dev/us-east-1/redis
    region: us-east-1
``` 

So this logic would have to be enhanced to parse multiple helmfiles.

Honestly, I'm not aware of what this looks like. I'll have to do my homework and learn `helmfile` a bit more.

Reply:
Let me look at this again. So this can run per `helmfile.yaml`